### DISCUSSION TITLE
Using Riemannian Adam on Hyperbolic Graph Neural Networks

### DISCUSSION THREAD

**Discussion by arijitthegame** [geoopt/geoopt#177] [General]
Hi everyone, 

 I am playing with the Hyperbolic Graph Neural Networks from Chami et al. and I am only interested in the hyperboloid model with different layers of HGCN having different trainable curvatures. I took their code and basically hardcoded the functions. Here is my notebook https://github.com/arijitthegame/Geometric-Deep-Learning/blob/master/Hyperbolic_GCN%20.ipynb 

However I am stuck with the optimizer. If I run it, I get an error saying "Dimension out of range (expected to be in range of [-1, 0], but got 1)" and tracing the error takes to expmap in line 148 and it is because for some reason the input to expmap is just one scalar. I am now confused as it is supposed to update all the weights. Also when I do : 

for point in group["params"]:
                  print(point)

It just prints out just one value of c and it is not clear which layer curvature it is. The way I have set it up, the curvature is a parameter of the model and I am not using the manifold class.

I apologize for this long post and will really appreciate any help.

**Comment by ferrine** [comment]
Hi, any questions still left? Sorry, But I totally missed reviewing this issue in time
